lib/src/main/java/org/kpipe/processor/AvroMessageProcessor.java

```diff
@@ -0,0 +1,682 @@
+    if (schema == null) {
+      throw new IllegalArgumentException("Schema not registered: " + schemaName);
+    }
+    return removeFields(schema, fields);
+  }
+
+  /**
+   * Creates a processor that removes specified fields from Avro data.
+   *
+   * <p>Example:
+   *
+   * <pre>{@code
+   * // Create a processor that removes sensitive fields
+   * Function<byte[], byte[]> sanitizeProcessor =
+   *     AvroMessageProcessor.removeFields(schema, "password", "creditCard", "ssn");
+   *
+   * // Register it in the registry
+   * MessageProcessorRegistry.register("sanitizeDataAvro", sanitizeProcessor);
+   *
+   * // Use it in a pipeline
+   * Function<byte[], byte[]> pipeline = MessageProcessorRegistry.pipeline(
+   *     "parseAvro", "sanitizeDataAvro", "addSourceAvro"
+   * );
+   *
+   * // Apply to a message
+   * byte[] inputAvro = ...; // Avro serialized bytes
+   * byte[] outputAvro = pipeline.apply(inputAvro);
+   * }</pre>
+   *
+   * @param schema The Avro schema to use
+   * @param fields Field names to remove
+   * @return Function that removes fields from Avro
+   */
+  public static Function<byte[], byte[]> removeFields(final Schema schema, final String... fields) {
+    final var fieldsToRemove = Set.of(fields);
+    return avroBytes -> {
+      if (avroBytes == null || avroBytes.length == 0) {
+        return EMPTY_AVRO;
+      }
+      try {
+        final var reader = new GenericDatumReader<GenericRecord>(schema);
+        final var inputStream = new ByteArrayInputStream(avroBytes);
+        final var record = reader.read(null, DecoderFactory.get().binaryDecoder(inputStream, null));
+        if (record == null) {
+          return EMPTY_AVRO;
+        }
+
+        final var newRecord = new GenericData.Record(schema);
+        schema
+          .getFields()
+          .forEach(field -> {
+            final var fieldName = field.name();
+            if (!fieldsToRemove.contains(fieldName)) {
+              newRecord.put(fieldName, record.get(fieldName));
+            } else {
+              final var fieldSchema = field.schema();
+              if (
+                fieldSchema.getType() == Schema.Type.UNION &&
+                fieldSchema.getTypes().stream().anyMatch(s -> s.getType() == Schema.Type.NULL)
+              ) {
+                newRecord.put(fieldName, null);
+              } else {
+                switch (fieldSchema.getType()) {
+                  case STRING:
+                    newRecord.put(fieldName, "");
+                    break;
+                  case INT:
+                  case LONG:
+                  case FLOAT:
+                  case DOUBLE:
+                    newRecord.put(fieldName, 0);
+                    break;
+                  case BOOLEAN:
+                    newRecord.put(fieldName, false);
+                    break;
+                  default:
+                    if (field.hasDefaultValue()) {
+                      newRecord.put(fieldName, field.defaultVal());
```

Comment:
should use default value if specified for Strings / numbers as well instead of "" or 0?

Reply:
will fix that